src/core/reactor.ts

```diff
@@ -2141,7 +2142,7 @@ export class App extends Reactor {
   /**
    * Priority set that keeps track of reactions at the current Logical time.
    */
-  private readonly _reactionQ = new ReactionQueue();
+  protected readonly _reactionQ = new ReactionQueue();
```

Comment:
Are you sure this is necessary? If so, why?

Reply:
It's used by `federatedApp`, a subclass of `Reactor`

 https://github.com/lf-lang/reactor-ts/blob/08fa08468d2c45714cd8e1fe6bd24f83edadda89/src/core/federation.ts#L1488-L1492

Reply:
@axmmisaka is right. Let me supplement it a little bit. 
From now on, `portAbsentReaction`s don't have a trigger. So we have to push those reactions to `_reactionQ` directly in the `FederateApp`. Also, I think this is safe because `_reactionQ` is owned by `App` and only `FederatedApp` is a subclass of `App`.

Reply:
OK, I agree that it's OK to make the visibility `protected`,

But if these reactions do not have a trigger, then how are they assigned a priority? Or are we adding these to the reaction queue only once we're ready to end of a tag, when all inputs are known and the reaction queue is empty?

Reply:
AFAIK, priorities are assigned to reactions, not triggers, right? Assuming that this is right, I think reactions are sorted by their priorities when they are pushed to `_reactionQ`. Currently, `portAbsentReaction`s are pushed into `_reactionQ` before executing reactions at each tag.

https://github.com/lf-lang/reactor-ts/blob/825f8fe3ea973c185be40de458583d3dcb00862b/src/core/reactor.ts#L2393-L2397

Reply:
It had escaped me that the reactions in the `NetworkSender` are code-generated. I'm assuming it works because the "port absent" reaction depends on the "port present" reaction that is triggered by the input? But can't they share the same trigger then? I haven't yet looked at the code generator carefully...

Reply:
>the "port absent" reaction depends on the "port present" reaction that is triggered by the input?

That's right. There is an edge that represents it in the `dependencyGraph`.
>But can't they share the same trigger then? 

For this question, I would share @petervdonovan's answer to the same question of mine. 

>That "port absent" reaction has to be triggered when msg is not present because its purpose is to send a message to notify the receiving federate that the port is absent in the current tag. So it would not work to have it be triggered by msg.